src/vm/gdbjit.cpp

```diff
@@ -2576,11 +2576,32 @@ void NotifyGdb::OnMethodPrepared(MethodDesc* methodDescPtr)
     }
 #endif
 
-    if (isListedModule(wszModuleFile))
+    // remove '.ni.dll' or '.ni.exe' suffix from wszModuleFile
+    size_t ModuleFileLen = wcslen(wszModuleFile);
+    LPWSTR wszNonNIModuleFile = new WCHAR[ModuleFileLen];
```

Comment:
You don't need to allocate a new memory for the string. You can just remove it in-place in the wszModuleName since the new string length is always smaller and nothing else uses the wszModuleName. It will allow you to significantly simplify the code. Like this:
```C++
LPCWSTR pNIExt = wcsstr(wszModuleFile, W(".ni.exe")); // where '.ni.exe' start at 
if (!pNIExt)
{
    pNIExt = wcsstr(wszModuleFile, W(".ni.dll")); // where '.ni.dll' start at 
}

if (pNIExt)
{
    wcscpy(pNIExt, W(".dll"));
}
```